### PR TITLE
add support for is_write_index in put-alias body parsing

### PR DESCRIPTION
**Pull request by talevy**
The Rest Put-Alias Action does separate parsing of the alias body
to construct the IndicesAliasesRequest. This extra parsing
was missed in #30703.

### CODE REVIEW

**Comment by talevy** [comment]
@elasticmachine test this please

File: rest-api-spec/src/main/resources/rest-api-spec/test/indices.put_alias/10_basic.yml [hunk 1]
```diff
@@ -56,3 +56,27 @@
...
+        index: test_index
+        name: test_alias
+        body:
+          is_write_index: true
+
+  - do:
+      indices.exists_alias:
+        name: test_alias
+
+  - is_true: ''
```
Comment (bleskes) [inline rest-api-spec/src/main/resources/rest-api-spec/test/indices.put_alias/10_basic.yml]: do we want to add a test that gets the alias and sees the flag?

**Review by bleskes — APPROVED** [review]
LGTM

**Comment by talevy** [comment]
thank you

**Comment by talevy** [comment]
I see. As is, this needs to be merged into 6.x first before master.

Per @rjernst's suggestion on managing bwc changes, I would like to do the following when 
I return to this.

1. merge this into master with bwc tests disabled
2. merge this as-is into 6.x
3. re-enable bwc tests on master

**Comment by rjernst** [comment]
> merge this as-is into 6.x

Note a very important part of this is to amend the cherry picked commit with the bwc tests *re-enabled*.

**Comment by talevy** [comment]
merged in 6.x here: https://github.com/elastic/elasticsearch/commit/cbd309bbb1dd5944e2d38a7eb9ba236676623110
and backcompat re-enabled here: https://github.com/elastic/elasticsearch/commit/60c1ea2c00356c347aad2b15ecbdcb11704c4a95

**Pull request merged**